packages/transaction-controller/src/utils/validation.ts

```diff
@@ -281,6 +286,14 @@ function validateGasFeeParams(txParams: TransactionParams) {
     );
     ensureFieldIsValidHex(txParams, 'maxPriorityFeePerGas');
   }
+
+  if (txParams.gasLimit) {
+    ensureFieldIsValidHex(txParams, 'gasLimit');
```

Comment:
Is there implications of this for existing dApps, do any rely on being able to provide decimals directly? A recall discussing this in the past.

Reply:
Maybe there are some but shouldn't we take this current API as base? 
https://docs.metamask.io/wallet/reference/json-rpc-methods/eth_sendtransaction/

Reply:
Yes, this is a good question, I don't want to break dApps with this update...

Reply:
Well, if I pass gas limit as a decimal number it isn't currently picked up by the UI. A different gas limit is shown. So I think it's safe to validate this as an hexadecimal.